platforms/software/dependency-management/src/main/java/org/gradle/internal/component/ConfigurationNotConsumableException.java

```diff
@@ -14,9 +14,14 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.exceptions;
+package org.gradle.internal.component;
```

Comment:
💭 This package seems a bit _too_ generic for all this stuff. Looking at it right now its not so cluttered so its probably fine to leave these here for now, but I think we should consider being more specific with all this selection stuff like putting it in `org.gradle.internal.component.selection` or something like that. 

There is a lot of code in the sub-packages of `org.gradle.internal.component`, and IMO it would make more sense for this code to live next to that code as opposed to above it. 

The kinds of things I would expect to live in `org.gradle.component` (and its internal counterpart):
`SoftwareComponent`
`SoftwareComponentVariant`
etc

i.e. -- things that define a component itself

Reply:
Absolutely.  I thought about making a subpackage for all the exceptions now, but figured I'd delay doing that until the message improvements were done, and then do it in a separate reorg PR.  I wanted to unify the hierarchy ahead of time here so I could also address these additional exception types now, so I just relocated them to have all the exceptions live together.